### PR TITLE
feat: display readable timestamp

### LINKED ISSUE #1888: Display `YYYY-MM-DD HH-MM-SS` format in the result of  `Show Components` command
The result of `Connect_time` is not convenient to debug
WX20220526-111212

### LINKED ISSUE #2416: display readable date and time for `show jobs`
currently it displays timestamp that is not human-readable, we should change it to normal readable data and time formats

### PR DESCRIPTION
**Pull request by dl239**
close #1888 #2416

### CODE REVIEW

File: src/base/time.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace openmldb {
+namespace base {
+
+constexpr int32_t TZ = 8;
+constexpr time_t TZ_OFFSET = TZ * 3600000;
+
+std::string Convert2FormatTime(int64_t ts) {
+    time_t time = (ts + TZ_OFFSET) / 1000;
+    struct tm t;
```
Comment (aceforeverd) [inline src/base/time.h:31]: memset t

File: src/sdk/result_set_sql.h [hunk 2]
```diff
@@ -91,20 +96,28 @@ class ResultSetSQL : public ::hybridse::sdk::ResultSet {
...
+    const bool GetAsString(uint32_t idx, std::string& val) override;
+
  private:
     ::hybridse::vm::Schema schema_;
     uint32_t record_cnt_;
     uint32_t buf_size_;
     std::shared_ptr<brpc::Controller> cntl_;
     ResultSetBase* result_set_base_;
     std::shared_ptr<butil::IOBuf> io_buf_;
+    bool readable_time_;
```
Comment (aceforeverd) [inline src/sdk/result_set_sql.h]: it's better that`ResultSet` should know nothing about readable time ?

Where prettify timestamp & date (or any other type) as string, appears in `sql_cmd.h`, with the customized data type formater provided

File: src/sdk/result_set_sql.cc [hunk 3]
```diff
@@ -113,33 +114,52 @@ std::shared_ptr<::hybridse::sdk::ResultSet> ResultSetSQL::MakeResultSet(
...
+    return MakeResultSet(schema, records, status);
+}
+
+const bool ReadableResultSetSQL::GetAsString(uint32_t idx, std::string& val) {
+    if (GetSchema()->GetColumnType(idx) == hybridse::sdk::kTypeTimestamp) {
+        int64_t ts = 0;
+        if (!GetTime(idx, &ts) || ts < 0) {
+            return false;
+        }
+        val = ::openmldb::base::Convert2FormatTime(ts);
```
Comment (dl239) [inline src/sdk/result_set_sql.cc]: date

**Review by vagetablechicken — APPROVED** [review]

File: src/sdk/CMakeLists.txt [hunk 4]
```diff
@@ -24,6 +24,9 @@ if(TESTING_ENABLE)
     add_executable(db_sdk_test db_sdk_test.cc)
     target_link_libraries(db_sdk_test base_test ${BIN_LIBS} ${THIRD_LIBS})
 
+    add_executable(result_set_sql_test result_set_sql_test.cc)
+    target_link_libraries(result_set_sql_test base_test ${BIN_LIBS} ${THIRD_LIBS})
```
Comment (aceforeverd) [inline src/sdk/CMakeLists.txt:28]: unit tests in `src/sdk` should append into list variable `test_list` so coverage will run it

Comment (aceforeverd) [reply src/sdk/CMakeLists.txt:28]: tracking in #2576

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**